### PR TITLE
[WIP] feat(default-theme): Add newsletter subscription

### LINKED ISSUE #1756: [FEATURE] Allow to unsubscribe an user from newsletter
### Description

There should be a place to allow to an user to unsubscribe provided email from newsletter.

### Acceptance criteria

- [ ] there is a dedicated page for such action, like another menu in "my account" for logged in users
- [ ] consider the feature to be available for guest customers too (if it make sense and it's possible to do so)
- [ ] POST `/newsletter/unsubscribe` is used to Remove a newsletter subscription for given email:
request's payload:
```json
{
  "email": "string"
}
```

### PR DESCRIPTION
**Pull request by mdanilowicz**
Work on this PR will be continued after backend fixes

## Changes
closes #1756

### Checklist

- [ ] the [list of features](docs/guide/FEATURELIST.md) is updated (if it's a feature and it's relevant)
- [ ] I followed [contributing](https://github.com/DivanteLtd/shopware-pwa/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: packages/default-theme/src/components/account/SwNewsletter.vue [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  <div class="newsletter-subcription">
+    <slot name="message">
+      <p class="message">
+        {{
+          $t("Here you can change your newsletter status")
+        }}
+      </p>
+    </slot>
+    <SwInput
+        v-model="user.email"
```
Comment (patzick) [inline packages/default-theme/src/components/account/SwNewsletter.vue:11]: we shouldn't modify the user's email here, for form purposes we can add prefilled ref, for example
```ts
const userEmail = ref(user.value?.email);
```

File: packages/default-theme/src/components/account/SwNewsletter.vue [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    >
+        {{ $t("Unsubscribe") }}
+    </SwButton>
+  </div>
+</template>
+
+<script>
+import { newsletterSubscribe, newsletterUnsubscribe } from "@shopware-pwa/shopware-6-client"
+import { getApplicationContext } from "@shopware-pwa/composables"
+import { useUser, useNotifications } from "@shopware-pwa/composables"
```
Comment (patzick) [inline packages/default-theme/src/components/account/SwNewsletter.vue:39]:
```suggestion
import { getApplicationContext, useUser, useNotifications } from "@shopware-pwa/composables"
```

**Review by patzick — CHANGES_REQUESTED** [review]
Very nice, just one thing to improve with form data model :)

**Comment by mkucmus** [comment]
@mdanilowicz  do we have an issue which blocks us here to link between repositories? available maybe here: https://issues.shopware.com

**Pull request closed without merging**